### ISSUE TITLE
pgcat does not honor startup parameters in transaction mode

### ISSUE THREAD

**Issue by drdrsh** [levkk/pgcat#72]
**Describe the bug**
I am comparing pgcat's behavior against pgbouncer with both running in transaction mode.

When the client sets any of the following parameters in the connection string, pgbouncer takes note of them and keeps them consistent for each client
```
application_name
client_encoding
datestyle
timezone
standard_conforming_strings
```

pgcat doesn't. In fact, it doesn't even set them when server connection is opened the first time. I think of these variables `application_name` is perhaps the most important given it can be used for query source attribution in `pg_stat_activity` 

**To Reproduce**
Run the following ruby script against pgcat after modifying the connection strings to point to the correct address
```
require 'pg'
$stdout.sync = true

conn1 = PG::connect("postgres://main_user:@pgcat:5432/main_db?application_name=app_1")
conn2 = PG::connect("postgres://main_user:@pgcat:5432/main_db?application_name=app_2")

puts "application_name on conn1 = #{conn1.async_exec('SHOW application_name')[0]['application_name']}"
puts "application_name on conn2 = #{conn2.async_exec('SHOW application_name')[0]['application_name']}"

conn1.close
conn2.close

```

This produces 

```
pgcat-report-app-1    | application_name on conn1 =
pgcat-report-app-1    | application_name on conn2 =
```

**Expected behavior**
It should output
```
pgcat-report-app-1    | application_name on conn1 =app_1
pgcat-report-app-1    | application_name on conn2 =app_2
```

**Comment by levkk** [comment]
Should be an easy fix, I'll push something soon @drdrsh 👍

**Comment by levkk** [comment]
Fixed `application_name`, the other ones I'll do later, they are slightly more involved I think. I'll keep this issue open until the remaining parameters are implemented.

**Comment by djohnson** [comment]
Hey @levkk, just stumbled across this open ticket while trying to debug some encoding issues.  We've seen temporary issues with client_encoding not being set that we suspect are related to this bug as they first started appearing after switching to pgcat.

**Comment by levkk** [comment]
Makes sense. You can set `client_encoding` on the server just like we do with `application_name`, e.g. https://github.com/levkk/pgcat/blob/master/src/client.rs#L1041

PR welcome!
